tests/trainer/optimization/test_optimizers.py

```diff
@@ -600,3 +602,21 @@ def test_optimizer_state_on_device(tmpdir):
         fast_dev_run=True,
     )
     trainer.fit(model)
+
+
+@pytest.mark.parametrize("check_val_every_n_epoch", [1, 2])
+@mock.patch("torch.optim.lr_scheduler.StepLR.step")
+def test_lr_scheduler_epoch_step_frequency(mocked_sched, check_val_every_n_epoch, tmpdir):
+    epochs = 4
+    expected_steps = epochs + 1  # every LRScheduler gets called once at init
+
+    model = BoringModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        limit_train_batches=2,
+        limit_val_batches=2,
+        check_val_every_n_epoch=check_val_every_n_epoch,
+        max_epochs=epochs,
+    )
+    trainer.fit(model)
+    assert mocked_sched.call_count == expected_steps
```

Comment:
noob question: why do we want to update schedulers all epochs even if we only run validation some of them?

Reply:
running validation on some epoch is just a choice. Ideally, a scheduler should update after every epoch if `scheduler['frequency] = 1(default)` and `scheduler['interval'] == 'epoch'`. If someone wants it to align it with validation, they can set `scheduler['frequency'] = check_val_every_n_epoch`.